proto/private/pbconfigentry/config_entry.gen.go

```diff
@@ -1834,7 +1834,7 @@ func RetryFilterFromStructs(t *structs.RetryFilter, s *RetryFilter) {
 	s.NumRetries = *t.NumRetries
 	s.RetryOn = t.RetryOn
 	s.RetryOnStatusCodes = t.RetryOnStatusCodes
-	s.RetryOnConnectFailure = *t.RetryOnConnectFailure
```

Comment:
Was the desire for this `RetryFilter.RetryOnConnectFailure` to originally have been 3-state? (enabled, disabled, default)

The protobuf form in `pbconfigentry` is flagged as `bool` and the structs internal form is `*bool` prior to John's change.

`mog` is generating a `bool <-> *bool` conversion, just like it is generating a `uint32 <-> *uint32` conversion 3 lines above.

I think there may be a `mog` bug in basic type pointer-to/from-nonpointer field conversions that might fix this differently. :eyes:

Reply:
I had a chat with John about this on the side, and the `.proto` isn't using the `optional` keyword. Although, it sounded like the crashes still happened even after that flag was introduced.

Reply:
If we land this mog thing the `*bool` behavior can come back: https://github.com/hashicorp/mog/pull/43